src/include/btmem.h

```diff
@@ -906,7 +906,10 @@ struct __wt_ref {
 #undef ref_ikey
 #define ref_ikey key.ikey
 
-    WT_PAGE_DELETED *page_del; /* Deleted page information */
+    union {                   /* Fast-truncate information */
+        WT_PAGE_DELETED *del; /* Not instantiated, page-deleted structure */
+        WT_UPDATE **update;   /* Instantiated, page updates for subsequent commit/abort */
+    } ft_info;
```

Comment:
I think the idea here is that when the `ref->state == WT_REF_DELETED`, the `del` pointer should be checked, but when `ref->state == WT_REF_MEM` (after instantiation), the `update` pointer should be checked.  If that is right, could we add a longer comment here explaining?

When an operation sets `ref->state = WT_REF_LOCKED`, the previous state should be used.  I don't think we should ever see either pointer set when the state is `WT_REF_DISK` or `WT_REF_SPLIT`, but let me know if I'm mistaken.

Reply:
Yes, that's correct: `WT_REF_DELETED` implies `WT_REF.del` is set, any in-memory `WT_REF` state implies `WT_REF.update` is possibly set. I believe `WT_REF_SPLIT` is possible if the page splits (IIRC, splits are what the `WT_REF.update` field is required for, otherwise we could find the WT_UPDATE structures using the txn's `WT_REF`), but any in-memory state should be OK. I don't think you can see `WT_REF_DISK`.

Change pushed in 0a57be59d for your review.